example/game/src/main/java/no/nordicsemi/android/ble/example/game/server/viewmodel/ServerViewModel.kt

```diff
@@ -0,0 +1,234 @@
+import no.nordicsemi.android.ble.example.game.server.repository.AdvertisingManager
+import no.nordicsemi.android.ble.example.game.server.repository.ServerConnection
+import no.nordicsemi.android.ble.example.game.server.repository.ServerManager
+import no.nordicsemi.android.ble.example.game.timer.TimerViewModel
+import no.nordicsemi.android.ble.ktx.state.ConnectionState
+import no.nordicsemi.android.ble.ktx.stateAsFlow
+import no.nordicsemi.android.ble.observer.ServerObserver
+import javax.inject.Inject
+
+@HiltViewModel
+class ServerViewModel @Inject constructor(
+    @ApplicationContext private val context: Context,
+    private val advertiser: AdvertisingManager,
+    private val serverManager: ServerManager,
+    private val questionRepository: QuestionRepository,
+) : TimerViewModel() {
+    var clients: MutableStateFlow<List<ServerConnection>> = MutableStateFlow(emptyList())
+    private val TAG = "Server Connection"
+    private var _state = MutableStateFlow<GameState>(WaitingForPlayers(0))
+    val state = _state.asStateFlow()
+    val isGameOver: MutableState<Boolean?> = mutableStateOf(null)
+
+    private val _selectedAnswer: MutableState<Int?> = mutableStateOf(null)
+    val selectedAnswer: State<Int?> = _selectedAnswer
+
+    private val _correctAnswerId: MutableStateFlow<Int?> = MutableStateFlow(null)
+    val correctAnswerId = _correctAnswerId.asStateFlow()
+    private var questionSaved: Questions? = null
+    var index = 0
+
+    private val totalQuestions: Int = questionSaved?.questions?.size ?: 10
+    val savedResult: MutableStateFlow<List<FinalResult>> = MutableStateFlow(emptyList())
+
+    init {
+        startServer()
+    }
+
+
+    fun startGame(category: Int? = null) {
+        advertiser.stopAdvertising()
+        isGameOver.value = false
+        index = 0
+
+        viewModelScope.launch {
+            _state.emit(DownloadingQuestions)
+            val questions = questionRepository.getQuestions(category = category)
+            questionSaved = questions
+            /** Send first Question */
+            showQuestion(questions.questions[0])
+        }
+    }
+
+    fun showNextQuestion() {
+        viewModelScope.launch {
+            questionSaved?.let { questions ->
+                index.takeIf { it + 1 < totalQuestions }
+                    ?.let { ++index }
+                    ?.let { questions.questions[it] }
+                    /** Send Next Question */
+                    ?.let { showQuestion(it) }
+                    ?: run {
+                        isGameOver.value = true
+                        clients.value.forEach {
+                            it.gameOver(ResultToClient(isGameOver.value!!, savedResult.value))
+                        }
+                    }
+            }
+        }
+    }
+
+    private suspend fun showQuestion(question: Question) {
+
+        clients.value.forEach {
+            if (isGameOver.value == false) {
```

Comment:
This check should be before forEach.

Reply:
Also, what if the game is over? Is it possible here? Why do you have to check it anyway, and if it is over, you're still starting countdown below.